internal/pkg/grpc/factory/factory.go

```diff
@@ -71,38 +82,54 @@ func NewDefaultOptions(setters ...Option) *Options {
 	return opts
 }
 
-// Listen configures TLS for mutual authtentication by loading the CA into a
-// CertPool and configuring the server's policy for TLS Client Authentication.
-// Once TLS is configured, the gRPC options are built to make use of the TLS
-// configuration and the receiver (Server) is registered to the gRPC server.
-// Finally the gRPC server is started.
-func Listen(r Registrator, setters ...Option) (err error) {
+// NewServer builds grpc server and binds it to the Registrator
```

Comment:
I did a bit of refactoring here so that I can do `Listen()` before `Serve()`.

The reason is that for `osd` or `trustd`, the only thing they're doing is serving the API, while `init` has some activity happening in main goroutine, while API should run in another goroutine.

Without the split, there are two problems:

1. When `factory.Listen()` runs in another goroutine, there's no way to tell whether API is ready, and so it's not clear whether directory for file socket has been created. Without that, it's tricky to say whether it's safe to bind-mount this directory when starting up `osd`.

2. If `factory.Listen()` runs in another goroutine, it's hard to say whether we should abort `init` or not on error - as it might be listener creation failure (should abort), or Serve failure (probably shouldn't).

New `ListenAndServe` == old `Listen` (name taken from net/http package).